src/ape/managers/networks.py

```diff
@@ -151,27 +155,36 @@ def network_choices(self) -> Iterator[str]:
             Iterator[str]: An iterator over all the network-choice possibilities.
         """
         for ecosystem_name, ecosystem in self.ecosystems.items():
+            if ecosystem_filter and ecosystem_name != ecosystem_filter:
+                continue
+
```

Comment:
Might be thinking about this wrong, but what if they type it incorrectly (i.e. etthereum)? Would we want a default here? Maybe an error message?

Reply:
- [ ] Validate ecosystem / network / provider names